node/caching-compiler.js

```diff
@@ -17,8 +17,17 @@ exports.compile = function (content, filename) {
     // Treat the REPL as if there was no filename.
     filename = null;
   }
-  var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
-  return info ? readWithCache(info, content) : content;
+  
+  try {
+    var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
+    return info ? readWithCache(info, content) : content;
+  } catch(e) {
+    if (filename) {
+      e.message += 'While processing file: ' + filename;
+    }
+    throw e;
+  }
+
 };
```

Comment:
Meteor's still on V8 4.5.103.43, but I agree this might be a premature optimization for one JS engine.